inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp

```diff
@@ -41,105 +41,81 @@ MKLDNNPlugin::MKLDNNInferRequest::~MKLDNNInferRequest() {
-            }
+    changeDefaultPtr();
 
-            InferenceEngine::TBlob<float> *in_f = nullptr;
-            switch (input.second->getTensorDesc().getPrecision()) {
-                case InferenceEngine::Precision::FP32:
-                    pushInput<float>(input.first, input.second);
-                    break;
-                case InferenceEngine::Precision::I32:
-                    pushInput<int32_t>(input.first, input.second);
-                    break;
-                case InferenceEngine::Precision::I8:
-                    pushInput<int8_t>(input.first, input.second);
-                    break;
-                case InferenceEngine::Precision::U16:
-                    // U16 is unsupported by mkldnn, so here we convert the blob and send I32
-                    copyConvert<int32_t>(InferenceEngine::Precision::I32, input, convertedInputs);
-                    break;
-                case InferenceEngine::Precision::I16:
-                    if (graph->hasMeanImageFor(input.first)) {
-                        // If a mean image exists, we convert the blob and send FP32
-                        copyConvert<float>(InferenceEngine::Precision::FP32, input, convertedInputs);
-                    } else {
-                        // Instead we can send I16 directly
-                        pushInput<int16_t>(input.first, input.second);
-                    }
-                    break;
-                case InferenceEngine::Precision::U8:
-                case InferenceEngine::Precision::BOOL:
-                    if (graph->hasMeanImageFor(input.first)) {
-                        // If a mean image exists, we convert the blob and send FP32
-                        copyConvert<float>(InferenceEngine::Precision::FP32, input, convertedInputs);
-                    } else if (dynamic_cast<InferenceEngine::TBlob<bool> *>(input.second.get())) {
-                        pushInput<bool>(input.first, input.second);
-                    } else {
-                        // Instead we can send I8 directly
-                        pushInput<uint8_t>(input.first, input.second);
-                    }
-                    break;
-                case InferenceEngine::Precision::I64:
-                    // I64 is unsupported by mkldnn, so here we convert the blob and send I32
-                    copyConvert<int32_t>(InferenceEngine::Precision::I32, input, convertedInputs);
-                    break;
-                case InferenceEngine::Precision::U64:
-                    // U64 is unsupported by mkldnn, so here we convert the blob and send I32
-                    copyConvert<int32_t>(InferenceEngine::Precision::I32, input, convertedInputs);
-                    break;
-                default:
-                    THROW_IE_EXCEPTION << "Unsupported input precision " << input.second->getTensorDesc().getPrecision();
-            }
-        }
-    }
+    // need to retain converted blobs until infer finish
```

Comment:
In fact I am not sure we really need. As far as I can see before your changes `convertedInputs` was destroyed up to `graph->Infer(m_curBatch);`